dungeon.js

```diff
@@ -1412,10 +1427,56 @@ function updateNumFloors() {
             data.push({"cells":[], "edges":[]});
         }
     }
+    
     FLOORS = newFloors;
     return true;
 }
 
+function updateSideLength() {
+    var sidelengthInput = document.getElementById("sidelength");
+    var newSideLength = sidelengthInput.value;
+    
+    if (newSideLength <= 0) {
+        console.error("Invalid map size!");
+        sidelengthInput.classList = ["input-error"];
+        return false;
+    } else {
+        sidelengthInput.classList = [];
+    }
+
+    if (newSideLength == HEIGHT) {
```

Comment:
what about `WIDTH` tho
I guess in practice it should always be the same, but to enforce that, maybe we should just have one variable?
(actually I think with this change if you manually tweaked the JSON to have a diff width/height it could work. but rotations might be broken for that)

Reply:
I thought about that, but in principle they could be separate variables in the future. Rotation is the biggest hurdle.
For now I assumed they would both be the same in all cases and have designed them as such.

Reply:
ok, seems fine to do this now and we can revisit rectangular maps in the future.. changing JSON by hand is kind of at your own risk anyways